### PR TITLE
update subgraph Protocol entity with new fields

### PR DESCRIPTION
**Pull request by adamsoffer**
**What does this pull request do? Explain your changes.**
This PR adds a new set of fields to the Protocol entity and introduces 4 new entities.

**Specific updates**
- Adds the following fields to the Protocol entity: 
`currentRound, inflation, inflationChange, lastInitializedRound, lastRoundLengthUpdateRound, lastRoundLengthUpdateStartBlock, lockPeriod, maxEarningsClaimsRounds, numActiveTranscoders, paused, roundLength, roundLockAmount, targetBondingRate, unbondingPeriod`
- Introduces the following new entities:
`SetCurrentRewardTokens, Pause, Unpause, ParameterUpdate`

**How did you test each of these updates**
The changes haven been deployed and synced on the Livepeer Canary subgraph

**Does this pull request close any open issues?**
#437 

**Checklist:**
- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/subgraph/src/mappings/livepeerToken.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
   let owner = event.params.owner
   let amount = event.params.value
   let spender = event.params.spender
 
-  let roundsManager = getRoundsManagerInstance(dataSource.network())
-  let currentRound = roundsManager.currentRound()
+  let protocol = Protocol.load('0')
+  if (protocol == null) {
+    protocol = new Protocol('0')
+  }
```
Comment (kyriediculous) [inline packages/subgraph/src/mappings/livepeerToken.ts]: We can shorten this to  `const protocol = Protocol.load('0') || new Protocol('0')` , that goes for every file in this PR where we do this. 

```
const a = null || undefined || "foo" 
console.log(a)
// "foo"
```

**Review by kyriediculous — CHANGES_REQUESTED** [review]
Looks good overall although my subgraph working knowledge is limited. One minor detail.

File: packages/subgraph/src/mappings/livepeerToken.ts (hunk 1 above)
Comment (adamsoffer) [reply packages/subgraph/src/mappings/livepeerToken.ts]: @kyriediculous yeah good call. I've updated each file with this shortened syntax.

**Comment by kyriediculous** [comment]
Changes look good, can we squash the commits on merge?

**Review by kyriediculous — APPROVED** [review]
